google/resource_cloudfunctions_function_test.go

```diff
@@ -0,0 +1,475 @@
+		switch n {
+		case FUNCTION_TRIGGER_HTTP:
+			if function.HttpsTrigger == nil {
+				return fmt.Errorf("Expected HttpsTrigger to be set")
+			}
+		case FUNCTION_TRIGGER_BUCKET:
+			if function.EventTrigger == nil {
+				return fmt.Errorf("Expected trigger_bucket to be set")
+			}
+			if strings.Index(function.EventTrigger.EventType, "cloud.storage") == -1 {
+				return fmt.Errorf("Expected cloud.storage EventType, found %s", function.EventTrigger.EventType)
+			}
+		case FUNCTION_TRIGGER_TOPIC:
+			if function.EventTrigger == nil {
+				return fmt.Errorf("Expected trigger_bucket to be set")
+			}
+			if strings.Index(function.EventTrigger.EventType, "cloud.pubsub") == -1 {
+				return fmt.Errorf("Expected cloud.pubsub EventType, found %s", function.EventTrigger.EventType)
+			}
+		default:
+			return fmt.Errorf("testAccCloudFunctionsFunctionTrigger expects only FUNCTION_TRIGGER_HTTP, " +
+				"FUNCTION_TRIGGER_BUCKET or FUNCTION_TRIGGER_TOPIC")
+		}
+		return nil
+	}
+}
+
+func testAccCloudFunctionsFunctionHasLabel(key, value string,
+	function *cloudfunctions.CloudFunction) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		val, ok := function.Labels[key]
+		if !ok {
+			return fmt.Errorf("Label with key %s not found", key)
+		}
+
+		if val != value {
+			return fmt.Errorf("Label value did not match for key %s: expected %s but found %s", key, value, val)
+		}
+		return nil
+	}
+}
+
+func createZIPArchiveForIndexJs(sourcePath string) (string, error) {
+	source, err := ioutil.ReadFile(sourcePath)
+	if err != nil {
+		return "", err
+	}
+	// Create a buffer to write our archive to.
+	buf := new(bytes.Buffer)
+
+	// Create a new zip archive.
+	w := zip.NewWriter(buf)
+
+	f, err := w.Create("index.js")
+	if err != nil {
+		return "", err
+	}
+	_, err = f.Write(source)
+	if err != nil {
+		return "", err
+	}
+
+	// Make sure to check the error on Close.
+	err = w.Close()
+	if err != nil {
+		return "", err
+	}
+	// Create temp file to write zip to
+	tmpfile, err := ioutil.TempFile("", "zip")
+	if err != nil {
+		return "", err
+	}
+
+	if _, err := tmpfile.Write(buf.Bytes()); err != nil {
+		return "", err
+	}
+	if err := tmpfile.Close(); err != nil {
+		return "", err
+	}
+	return tmpfile.Name(), nil
+}
+
+func testAccCloudFunctionsFunction_basic(functionName string, bucketName string, zipFilePath string) string {
+	return fmt.Sprintf(`
+resource "google_storage_bucket" "bucket" {
+  name = "%s"
+}
+
+resource "google_storage_bucket_object" "archive" {
+  name   = "index.zip"
+  bucket = "${google_storage_bucket.bucket.name}"
+  source = "%s"
+}
+
+resource "google_cloudfunctions_function" "function" {
+  name                  = "%s"
+  description           = "test function"
+  available_memory_mb   = 128
+  source_archive_bucket = "${google_storage_bucket.bucket.name}"
+  source_archive_object = "${google_storage_bucket_object.archive.name}"
+  trigger_http          = true
+  timeout		        = 61
```

Comment:
nit: missed one (alignment)

Reply:
Fixed, sorry for this - my editor didn't show those tabs...